src/FSharp.Control.TaskSeq/TaskSeq.fsi

```diff
@@ -154,414 +175,872 @@ module TaskSeq =
+    /// <summary>
+    /// Views the given <see cref="list" /> of <see cref="Task" />s as a task sequence, that is, as an
+    /// <see cref="IAsyncEnumerable&lt;'T>" />. A list of tasks will typically already be hot-started,
+    /// as a result, each task can already run and potentially out of order, or with
+    /// overlapping side effects, while a task sequence forces awaiting between the items in the sequence,
+    /// preventing such overlap to happen. Converting a list of tasks into a task sequence is no guarantee
+    /// that overlapping side effects are prevented. Safe for side-effect free tasks.
+    /// </summary>
+    ///
+    /// <param name="source">The input list-of-tasks.</param>
+    /// <returns>The resulting task sequence.</returns>
     val ofTaskList: source: #Task<'T> list -> taskSeq<'T>
 
-    /// Create a taskSeq of an array of tasks, that may already have hot-started.
+    /// <summary>
+    /// Views the given <see cref="array" /> of <see cref="Task" />s as a task sequence, that is, as an
+    /// <see cref="IAsyncEnumerable&lt;'T>" />. An array of tasks will typically already be hot-started,
+    /// as a result, each task can already run and potentially out of order, or with
+    /// overlapping side effects, while a task sequence forces awaiting between the items in the sequence,
+    /// preventing such overlap to happen. Converting an array of tasks into a task sequence is no guarantee
+    /// that overlapping side effects are prevented. Safe for side-effect free tasks.
+    /// </summary>
+    ///
+    /// <param name="source">The input array-of-tasks.</param>
+    /// <returns>The resulting task sequence.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input array is null.</exception>
     val ofTaskArray: source: #Task<'T> array -> taskSeq<'T>
 
-    /// Create a taskSeq of a seq of async.
+    /// <summary>
+    /// Views the given <see cref="seq" /> of <see cref="Async" />s as a task sequence, that is, as an
+    /// <see cref="IAsyncEnumerable&lt;'T>" />. A sequence of asyncs is not the same as a task sequence.
+    /// Each async computation in a sequence of asyncs can be run individually or in parallel, potentially
+    /// with overlapping side effects, while a task sequence forces awaiting between the items in the sequence,
+    /// preventing such overlap to happen.
+    /// </summary>
+    ///
+    /// <param name="source">The input sequence-of-asyncs.</param>
+    /// <returns>The resulting task sequence.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input sequence is null.</exception>
     val ofAsyncSeq: source: seq<Async<'T>> -> taskSeq<'T>
 
-    /// Create a taskSeq of a list of async.
+    /// <summary>
+    /// Views the given <see cref="list" /> of <see cref="Async" />s as a task sequence, that is, as an
+    /// <see cref="IAsyncEnumerable&lt;'T>" />. A list of asyncs is not the same as a task sequence.
```

Comment:
here the qualifier self-discussion I delve into in my comment on ofTaskSeq is not warranted, and this is clear and unambiguous

Reply:
@bartelink can you comment these further on in #136? It was my mistake to base this branch off of that one, but this should really only be about the two added groups of functions.

Reply:
PS: no need to repeat, I'll take your suggestions to heart in that PR.